src/components/popover/popover.jsx

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import IconButton from '../icon-button/icon-button.jsx';
+import iconClear from '../../svgs/clear.svg';
+import styles from './popover.css';
+
+const Popover = forwardRef(function Popover(props, ref) {
+  const [open, setOpen] = useState(props.startOpen);
+  const closeIconRef = useRef();
+  const backupRef = useRef();
+  const sectionRef = useRef();
+  const selfRef = ref || backupRef;
+
+  const onFocusIn = (event) => {
+    if (open && event.target instanceof HTMLElement && !sectionRef.current.contains(event.target)) {
```

Comment:
This is a first I'm seeing a type check on event.target. Is it possible to get something without `contains`?

The class inheritance looks like:

EventTarget -> Node -> Element -> HTMLElement

And the `contains` method is defined on the Node class. 

> Element, Document, and Window are the most common event targets, but other objects can be event targets, too. For example XMLHttpRequest, AudioNode, AudioContext, and others.

Did you see something without a `contains` while developing?

Reply:
@juanca while testing, actually! Sometimes in testing, there seems to be `focus` events happening where `event.target` is not `null || undefined`, but is also not `Node`. 😬